src/workerd/jsg/modules.h

```diff
@@ -612,10 +612,9 @@ v8::MaybeLocal<v8::Promise> dynamicImportCallback(v8::Local<v8::Context> context
                                                   v8::Local<v8::Value> resource_name,
                                                   v8::Local<v8::String> specifier,
                                                   v8::Local<v8::FixedArray> import_assertions) {
-  auto isolate = context->GetIsolate();
-  auto& lock = Lock::from(isolate);
-  auto registry = ModuleRegistry::from(lock);
-  auto& wrapper = TypeWrapper::from(isolate);
+  auto& js = Lock::from(context->GetIsolate());
```

Comment:
Not new in this PR, but it appears `dynamicImportCallback` is called by V8, which means the body needs to be wrapped in something like `liftKj` or a try/catch to make sure we don't throw exceptions back to V8.

Could fix in a separate PR if preferred.